kubedog.go

```diff
@@ -66,6 +66,14 @@ func (kdt *Test) Run() {
 	kdt.scenarioContext.Step(`^an Auto Scaling Group named ([^"]*)$`, kdt.AwsContext.AnASGNamed)
 	kdt.scenarioContext.Step(`^(?:I )?update (?:the )current Auto Scaling Group with ([^"]*) set to ([^"]*)$`, kdt.AwsContext.UpdateFieldOfCurrentASG)
 	kdt.scenarioContext.Step(`(?:the )?current Auto Scaling Group (?:is )scaled to \(min, max\) = \((\d+), (\d+)\)$`, kdt.AwsContext.ScaleCurrentASG)
+	kdt.scenarioContext.Step(`^(?:I )?print (?:the )?nodes list$`, kdt.KubeContext.PrintNodes)
+	kdt.scenarioContext.Step(`^(?:I )?print (?:the )?pods in namespace ([^"]*) with selector ([^"]*)$`, kdt.KubeContext.PrintPodsWithSelector)
+	kdt.scenarioContext.Step(`^(?:I )?print (?:the )?pods in namespace ([^"]*)$`, kdt.KubeContext.PrintPods)
+	kdt.scenarioContext.Step(`^(?:the )?deployment ([^"]*) is running in namespace ([^"]*)$`, kdt.KubeContext.DeploymentIsRunning)
+	kdt.scenarioContext.Step(`^(?:the )?daemonset ([^"]*) is running in namespace ([^"]*)$`, kdt.KubeContext.DaemonsetIsRunning)
+	kdt.scenarioContext.Step(`^(?:the )?persistentvolume ([^"]*) exists with status (Available|Bound|Released|Failed|Pending)$`, kdt.KubeContext.PersistentVolExists)
+	kdt.scenarioContext.Step(`^(?:I )?(add|remove) efs-csi-role as trusted entity to iam role ([^"]*)$`, kdt.KubeContext.EfsCsiRoleTrust)
```

Comment:
I think this step can be made more generic, but the efs-csi-role step will need to stay separate. The role arn is actually `fmt.Sprintf("arn:aws:iam::%s:role/efs-csi-role-%s", accountId, BDDClusterName)`, but I can make a separate step that does the same thing but does `fmt.Sprintf("arn:aws:iam::%s:role/%s", accountId, roleName)` instead. efs-csi-role needs to adhere to a specific naming scheme and we can't put the full name into the .feature file because it will only work on a specific cluster name then.

Reply:
Added a generic step